doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,574 @@
+
+- SupportedIndicesMap
+    .. code-block:: go
+
+        type ConnectionPoint struct {
+            Type        string
+            IP          string
+            Prefix      uint32
+            InterfaceID uint16
+        }
+
+    Maps a connection pair consisting of two ConnectionPoints to a list of policy indices.
+    This map shows for each connection pair which policy indices are supported, which can be one or multiple policies.
+    A ConnectionPoint is either an interface, an IP range or wildcard.
+    For all intermediary hops interface to interface connection points will be used whereas interface to IP range is used for the last hop.
+- IndexIdentifierMap
+    A policy index is to be embedded in the HBH extension and therefore has to be minimal in size.
+    The size of a policy index is 8 bits, whereas identifiers can be a multiple of this (especially global identifiers).
+    The policy index is thus different to the policy identifier. In order to decode which policies are supported on which interfaces,
+    a mapping is required from policy index to local and global identifiers.
+    This mapping is provided by this map.
+- IdentifierDescriptionMap
+    Global identifiers can be found in a global datastore, but local identifiers are specific to an AS.
+    This map maps a local policy identifier to its corresponding description.
+- MPLSMaps
+    Routers need to be aware of the supported policy indices and the corresponding MPLS config they need to apply to packets to
+    enforce the policy in the internal network.
+    Routers periodically fetch this map from the control service.
+    A hash of the MPLS map is maintained, such that routers only have to update if their hash differs from the one at the control service.
+- RemotePolicyCache
+    When a local policy is queried at a remote AS, the resulting policy description is cached at the requesting AS' FABRID Manager,
+    such that subsequent requests can be served from cache.
+
+
+PCB dissemination
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The *IndexIdentifierMap* and *SupportedIndicesMap* are included in a (unsigned) detachable extension in the PCBs for an AS.
+Hashes of these maps are maintained in a Signed AS Entry, such that the authenticity of these maps can be verified.
+The detachable extension can also be present in the PCB, i.e. it does not have to be detached in all cases, e.g. if there are only very few policies.
+If the maps are detached, they can be fetched from the control service of that AS and the received maps can be verified with the hashes.
+To ensure a consistent hash calculation, the key entries of these maps have to be sorted, such that they are accessed in a consistent order.
+
+Exposing policies to the end hosts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. _endhost_policy_selection:
+The path combinator finds the most recent FABRID map per AS among the received segments and subsequently uses this map to find the FABRID
```

Comment:
What if an end host wants to fetch FABRID policies from an AS that does not support FABRID? What if the service supports FABRID, but is down?

In such cases, how long will the end host wait until it stops trying to request the policies?

Reply:
Also, is it possible for an end host to fetch all global policies and their human-readable descriptions?
Similarly, is it possible for an end host to list the descriptions of all local policies of a specific AS (or maybe even for all on-path ASes) in human readable format?

Reply:
> What if an end host wants to fetch FABRID policies from an AS that does not support FABRID? What if the service supports FABRID, but is down?
> 
> In such cases, how long will the end host wait until it stops trying to request the policies?

The end host gets the FABRID policies alongside the paths that it has requested. For ASes that do not yet support FABRID it will return an empty list for the FABRID policies, there is no delay here. If an end host decides to query for policies at an AS that does not support FABRID, there is a timeout of a few seconds. What should be done after the timeout is up to the application I would say. 

> Also, is it possible for an end host to fetch all global policies and their human-readable descriptions?

No, the global policies are assumed to be in some sort of 'external' list. The list is not implemented.

> Similarly, is it possible for an end host to list the descriptions of all local policies of a specific AS (or maybe even for all on-path ASes) in human readable format?

We currently only have an endpoint to query for a specific local policy. This can be easily extended to all policies or a set of policies.

Reply:
- For the first part, could you please put this explanation into the design document?
- For the global policies, would be good to have an append-only list somewhere, e.g., the scionlab github repo?
- For the policy query, will the end host be able to see the actual textual description of the policy, or just the index/identifier?

Reply:
Regarding the third point, the end host will be able to see the textual description.

Reply:
Thanks for the clarifications!

@rohrerj : I would suggest to add the following:
- "If an end host decides to query for policies at an AS that does not support FABRID, there is a timeout of a few seconds. What should be done after the timeout is up to the application."
 - Mention here that the end host will be able to see the textual policy description (not just an index or identifier).
- Mention that the global policy list is not implemented yet.
- Add another paragraph to the "future work" section regarding having an append-only list somewhere, e.g., the scionlab github repo.

Reply:
Done